failpoints_test.go

```diff
@@ -1,158 +1,171 @@
 	start := time.Now()
 	v, err := fps.Eval("gofail/testPause")
-	c.Assert(err, IsNil)
-	c.Assert(v, IsNil)
-	c.Assert(time.Since(start), GreaterEqual, 100*time.Millisecond, Commentf("not paused"))
+	require.NoError(t, err)
+	require.Nil(t, v)
+
+	require.True(t, time.Since(start) > 100*time.Millisecond)
 	<-ch
 
 	err = fps.Enable("failpoints-test-4", "50.0%return(5)")
-	c.Assert(err, IsNil)
+	require.NoError(t, err)
 	var succ int
 	for i := 0; i < 1000; i++ {
 		val, err = fps.Eval("failpoints-test-4")
 		if err == nil && val != nil {
 			succ++
-			c.Assert(val.(int), Equals, 5)
+			require.Equal(t, 5, val.(int))
 		}
 	}
+
 	if succ < 450 || succ > 550 {
-		c.Fatalf("prop failure: %v", succ)
+		require.FailNow(t, "prop failure: %v", succ)
 	}
 
 	err = fps.Enable("failpoints-test-5", "50*return(5)")
-	c.Assert(err, IsNil)
+	require.NoError(t, err)
 	for i := 0; i < 50; i++ {
 		val, err = fps.Eval("failpoints-test-5")
-		c.Assert(err, IsNil)
-		c.Assert(val.(int), Equals, 5)
+		require.NoError(t, err)
+		require.Equal(t, 5, val.(int))
 	}
 	val, err = fps.Eval("failpoints-test-5")
-	c.Assert(errors.Cause(err), Equals, failpoint.ErrNotAllowed)
-	c.Assert(val, IsNil)
+	require.Equal(t, failpoint.ErrNotAllowed, errors.Cause(err))
+	require.Nil(t, val)
 
 	points := map[string]struct{}{}
 	for _, fp := range fps.List() {
 		points[fp] = struct{}{}
 	}
-	c.Assert(points, HasKey, "failpoints-test-1")
-	c.Assert(points, HasKey, "failpoints-test-2")
-	c.Assert(points, HasKey, "failpoints-test-3")
-	c.Assert(points, HasKey, "failpoints-test-4")
-	c.Assert(points, HasKey, "failpoints-test-5")
+	require.Contains(t, points, "failpoints-test-1")
+	require.Contains(t, points, "failpoints-test-2")
+	require.Contains(t, points, "failpoints-test-3")
+	require.Contains(t, points, "failpoints-test-4")
+	require.Contains(t, points, "failpoints-test-5")
 
 	err = fps.Enable("failpoints-test-6", "50*return(5)->1*return(true)->1*return(false)->10*return(20)")
-	c.Assert(err, IsNil)
+	require.NoError(t, err)
 	// 50*return(5)
 	for i := 0; i < 50; i++ {
 		val, err = fps.Eval("failpoints-test-6")
-		c.Assert(err, IsNil)
-		c.Assert(val.(int), Equals, 5)
+		require.NoError(t, err)
+		require.Equal(t, 5, val.(int))
 	}
 	// 1*return(true)
 	val, err = fps.Eval("failpoints-test-6")
-	c.Assert(err, IsNil)
-	c.Assert(val.(bool), IsTrue)
+	require.NoError(t, err)
+	require.True(t, val.(bool))
 	// 1*return(false)
 	val, err = fps.Eval("failpoints-test-6")
-	c.Assert(err, IsNil)
-	c.Assert(val.(bool), IsFalse)
+	require.NoError(t, err)
+	require.False(t, val.(bool))
 	// 10*return(20)
 	for i := 0; i < 10; i++ {
 		val, err = fps.Eval("failpoints-test-6")
-		c.Assert(err, IsNil)
-		c.Assert(val.(int), Equals, 20)
+		require.NoError(t, err)
+		require.Equal(t, 20, val.(int))
 	}
 	val, err = fps.Eval("failpoints-test-6")
-	c.Assert(errors.Cause(err), Equals, failpoint.ErrNotAllowed)
-	c.Assert(val, IsNil)
+	require.Equal(t, failpoint.ErrNotAllowed, errors.Cause(err))
```

Comment:
```suggestion
	require.ErrorIs(t, failpoint.ErrNotAllowed, err)
```

Reply:
It seems these two checkers have different logic ([report](https://github.com/pingcap/failpoint/pull/68/checks?check_run_id=3639730679)). I'm going to revert these changes. pingcap/errors is not totally compatible with go errors.

Reply:
oops 😂